src/workerd/io/hibernation-manager.h

```diff
@@ -187,6 +187,11 @@ class HibernationManagerImpl final : public Worker::Actor::HibernationManager {
     TagCollection(TagCollection&& other) = default;
   };
 
+  struct AutoRequestResponsePair {
```

Comment:
non-blocking: If we changed this from 2 independent Maybe's to a Maybe<pair> then it would cleanup most of the other state stuff I mentioned.